edge/pkg/servicebus/servicebus.go

```diff
@@ -88,72 +121,154 @@ func processMessage(msg *beehiveModel.Message) {
-	resBody, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		if err.Error() == "http: request body too large" {
-			err = fmt.Errorf("response body too large")
+
+		//send message with resource to the edge part
+		operation := httpRequest.Method
+		targetURL := "http://127.0.0.1:" + r[0] + r[1]
+		resp, err := uc.HTTPDo(operation, targetURL, httpRequest.Header, httpRequest.Body)
+		if err != nil {
+			m := "error to call service"
+			code := http.StatusNotFound
+			klog.Errorf(m, err)
+			if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
+				beehiveContext.SendToGroup(modules.HubGroup, response)
+			}
+			return
 		}
-		m := "error to receive response, err: " + err.Error()
-		code := http.StatusInternalServerError
-		klog.Errorf(m, err)
-		if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
-			beehiveContext.SendToGroup(modules.HubGroup, response)
+		defer resp.Body.Close()
+		resp.Body = http.MaxBytesReader(nil, resp.Body, maxBodySize)
+		resBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			if err.Error() == "http: request body too large" {
+				err = fmt.Errorf("response body too large")
+			}
+			m := "error to receive response, err: " + err.Error()
+			code := http.StatusInternalServerError
+			klog.Errorf(m, err)
+			if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
+				beehiveContext.SendToGroup(modules.HubGroup, response)
+			}
+			return
 		}
-		return
+
+		response := commonType.HTTPResponse{Header: resp.Header, StatusCode: resp.StatusCode, Body: resBody}
+		responseMsg := beehiveModel.NewMessage(msg.GetID())
+		responseMsg.Content = response
+		responseMsg.SetRoute("servicebus", modules.UserGroup).SetResourceOperation("", beehiveModel.UploadOperation)
+		beehiveContext.SendToGroup(modules.HubGroup, *responseMsg)
+	}
+}
+
+func server(stopChan <-chan struct{}) {
+	var (
+		timeout time.Duration
+		err     error
+	)
+	if timeout, err = time.ParseDuration(fmt.Sprintf("%vs", servicebusConfig.Config.Timeout)); err != nil {
+		klog.Errorf("can't format timeout and the default value will be setted")
+		timeout, _ = time.ParseDuration("10s")
 	}
 
-	response := commonType.HTTPResponse{Header: resp.Header, StatusCode: resp.StatusCode, Body: resBody}
-	responseMsg := beehiveModel.NewMessage(msg.GetID())
-	responseMsg.Content = response
-	responseMsg.SetRoute("servicebus", modules.UserGroup).SetResourceOperation("", beehiveModel.UploadOperation)
-	beehiveContext.SendToGroup(modules.HubGroup, *responseMsg)
+	h := buildBasicHandler(timeout)
+	s := http.Server{
+		Addr:    fmt.Sprintf("%s:%d", servicebusConfig.Config.Server, servicebusConfig.Config.Port),
+		Handler: h,
+	}
+	go func() {
+		<-stopChan
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		if err := s.Shutdown(ctx); err != nil {
+			klog.Errorf("Server shutdown failed: %s", err)
+		}
+		inited = 0
```

Comment:
whether do we need to use `atomic.CompareAndSwapInt32` method here, and I doubt if we need to use atomic methods to process the int variable `inited`

Reply:
yeap you are right.